src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java

```diff
@@ -90,81 +128,95 @@ private void removeHoldingUnrecognizedFields(List<JsonObject> holdings) {
+      .compose(newHoldingId -> {
+        holder.addHoldingRefsToStoragePatchOrderLineRequest(holdingId, newHoldingId);
+        return updateItemsHolding(holdingId, newHoldingId, poLine.getId(), locationContext);
+      });
+  }
+
+  private Future<List<Location>> retrieveUniqueLocations(PoLine poLine, RequestContext requestContext) {
+    return pieceStorageService.getPiecesByPoLineId(PoLineCommonUtil.convertToCompositePoLine(poLine), requestContext)
       .map(pieces -> {
         List<Location> pieceHoldingIds = pieces
           .stream()
           .map(piece -> new Location().withHoldingId(piece.getHoldingId()).withLocationId(piece.getLocationId()))
           .collect(toList());
-        List<Location> storageHoldingIds = holder.getStoragePoLine().getLocations();
+        List<Location> storageHoldingIds = poLine.getLocations();
 
-        StreamEx.of(ListUtils.union(pieceHoldingIds, storageHoldingIds))
+        return StreamEx.of(ListUtils.union(pieceHoldingIds, storageHoldingIds))
           .distinct(location -> String.format("%s %s", location.getLocationId(), location.getHoldingId()))
           .filter(location -> Objects.nonNull(location.getHoldingId()))
-          .forEach(consumer);
-        return null;
-      })
-      .map(pieces -> null);
+          .toList();
+      });
   }
 
   private Future<Void> updateItemsHolding(String holdingId, String newHoldingId, String poLineId, RequestContext requestContext) {
     return inventoryItemManager.getItemsByHoldingIdAndOrderLineId(holdingId, poLineId, requestContext)
-        .map(items -> updateItemHoldingId(items, newHoldingId))
-        .compose(items -> updateItemsInInventory(items, requestContext));
-  }
-
-  private List<JsonObject> updateItemHoldingId(List<JsonObject> items, String holdingId) {
-    items.forEach(item -> item.put(ITEM_HOLDINGS_RECORD_ID, holdingId));
-    return items;
+      .compose(items -> updateItemsInInventory(items, newHoldingId, requestContext));
   }
 
-  private Future<Void> updateItemsInInventory(List<JsonObject> items, RequestContext requestContext) {
+  private Future<Void> updateItemsInInventory(List<JsonObject> items, String newHoldingId, RequestContext requestContext) {
+    items.forEach(item -> item.put(ITEM_HOLDINGS_RECORD_ID, newHoldingId));
     List<Parameter> parameters = new ArrayList<>();
     return GenericCompositeFuture.join(items.stream()
-      .map(item -> inventoryItemManager.updateItem(item, requestContext)
-        .otherwise(ex -> {
-          Parameter parameter = new Parameter().withKey("itemId").withValue(item.getString(ID));
-          if (ex.getCause() instanceof HttpException httpException) {
-            parameter.withAdditionalProperty("originalError", httpException.getError().getMessage());
-          }
-          parameters.add(parameter);
-          return null;
-        }))
-      .collect(toList()))
+        .map(item -> inventoryItemManager.updateItem(item, requestContext)
```

Comment:
previous formatting with 2 spaces was correct

Reply:
My IDE thinks differently, can you please double-check?

Reply:
please put items or stream to the new line, so it will be more clear

Reply:
https://github.com/folio-org/mod-orders/pull/946/files/cbb359a107b1f21dbfac8bd84db427bc9032c65d#diff-10b2663c3e84a0c17382f2ec0a59b146dbb7e907a02c109590650de87d88581aL157

Reply:
done